<launch>
	<arg name="robot_ns" default=""/>
	<arg name="sim" default=""/>

	<rosparam file="$(find sd_sensor)/config/$(arg robot_ns)_sensor_config.yaml" command="load"/>
	
	<!--R2 sonar are not operationnals
	node pkg="sd_sensor" type="$(arg robot_ns)_sensor_node" name="$(arg robot_ns)_sensor_node" if="$(eval arg('robot_ns') == 'r2')">
	</node-->

	<node name="mpu6050"  pkg="sd_sensor"  type="mpu6050_fifo_node" output="screen" respawn="false" if="$(eval not arg('sim'))">
		<remap from="imu/data" to="imu/data_raw"/>
	</node>

	<node name="ydlidar"  pkg="ydlidar"  type="ydlidar_node" output="screen" respawn="false" if="$(eval arg('robot_ns') == 'r1' and not arg('sim'))">
		<remap from="scan" to="laser/scan" />
	</node>

	<node name="xv11lidar" pkg="xv_11_lidar_raspberry" type="xv_11_lidar_raspberry" output="screen" if="$(eval arg('robot_ns') == 'r2' and not arg('sim'))">
		<param name="port" value="/dev/ttyAMA0"/>
		<param name="frame_id" value="r2/lidar_link"/>
		<remap from="scan" to="laser/scan"/>
	</node>

	<node pkg="laser_filters" type="scan_to_scan_filter_chain" name="$(arg robot_ns)_laser_scan_filter" output="screen">
		<remap from="scan" to="laser/scan" />
		<remap from="scan_filtered" to="laser/scan_filtered" />
	</node>

	<node pkg="sd_sensor" type="laser_pattern_detector" name="$(arg robot_ns)_laser_pattern_detector" output="screen">
		<remap from="scan" to="laser/scan" />
	</node>

</launch>